[docs] Add a support page (#17437)
mui-org/material-ui: docs/src/pages/getting-started/support/support.md

#### New issue guidelines

- Please follow the issue template.
- Please begin the title with '[ComponentName]' where appropriate, and use a succinct description that helps others find similar issues.
  - "doesn't work" ❌
  - "support date range picker" ✅
- Please don't group multiple topics into one issue, but instead, each should be its own issue.
- Please don't comment "+1" on an issue. It spams the maintainers and doesn't help move the issue forward. Use GitHub reactions instead (👍).

### Spectrum <svg style="width: 24px; fill: currentColor; margin-left: 0px;" fill-rule="evenodd" clip-rule="evenodd" stroke-linejoin="round" stroke-miterlimit="1.414" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32" preserveAspectRatio="xMidYMid meet"><path d="M6,14.5c0,0.828 0.672,1.5 1.5,1.5l1.5,0c3.866,0 7,3.134 7,7l0,1.5c0,0.828 0.672,1.5 1.5,1.5l7,0c0.828,0 1.5,-0.672 1.5,-1.5l0,-1.5c0,-9.389 -7.611,-17 -17,-17l-1.5,0c-0.828,0 -1.5,0.672 -1.5,1.5l0,7Z"></path></svg>

Chat with the community on Spectrum. If you feel your thread could be a StackOverflow question, go to StackOverflow instead. We lock/delete questions that are better asked on StackOverflow to avoid duplicates.

> [Start a discussion](https://spectrum.chat/material-ui)

Author: It has been useful on a few occasions in the past. For instance, it has helped to classify concerns between question and issue. I would be curious to see how it gets used with this new documentation structure. If we still get a majority of questions, that should be asked on StackOverflow, within a month, I think that we should consider doing something. What do you think? I have looked at the performance recently.

### Blog 📝

Stay up to date with the development of the library by following the blog.

[Read the latest posts](https://medium.com/material-ui/)

### Resources 📖

New to Material-UI? It's easy to learn if you know where to start!

[Learn Material-UI](/getting-started/learn/)

### Twitter <svg style="width: 24px; fill: currentColor; margin-left: 4px;" role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z" /></svg>

Receive the latest news on Material-UI.

[Follow us](https://twitter.com/MaterialUI)

### Supported versions

> The Material-UI CE (Community Edition) has been 100% open-source (MIT) since the very beginning, and always will be. Developers can ensure Material-UI is the right choice for their React applications through Material-UI’s community maintenance strategy. The Material-UI team regularly ships new releases, bug fixes, and is very welcoming to community pull requests.

Author: I think that we should start to seed the diff between the MIT and non-MIT components. I think that introducing the notion of community edition is important. I have updated the roadmap in https://github.com/mui-org/material-ui/pull/17439/files to start to shape what will build as paid.

Given the reality of time and resource restraints as well as the desire to keep innovating in the frontend development space, over time it becomes necessary for the Material-UI team to shift focus to newer versions of the framework. However, Material-UI will do everything it can to make the transition to newer versions as smooth as possible, including but not limited to publishing migration guides such as [this one for Material-UI v4](/guides/migration-v3/). The open-source community is always welcome to submit new features and bug fixes as well.

The current status of each Material-UI framework version is:

- Material-UI v4 (hooks) **[Active]**: Latest version in active development.
- Material-UI v3 (browser version support): Only handle security fixes.
- Material-UI v2 (never existed).
- Material-UI v1 (rewrite): Inactive.
- Material-UI v0.x: Inactive.

For teams and organizations that require additional support for any framework version, Material-UI has [options available](#enterprise).

> ## Professional help (Premium)

Reviewer: ```suggestion
## Professional support (Premium)
```

### Enterprise

Accelerate your company’s application development with a secure, long-term solution that’s optimized for enterprise and backed by Material-UI experts.

- **Long Term Support.** ([LTS](https://en.wikipedia.org/wiki/Long-term_support)) provides developers and organizations with additional stability and reliability in the applications they’re developing, including guaranteed support level agreements (SLAs) and advisory services. Take the preparation time you need before upgrading to the latest active version (v4) of Material-UI or remain confident in your application’s long-term success by staying on whichever version you’re most comfortable with. LTS is available as a paid service, it extends to Material-UI v1 implementations and higher.
- **Version Migration Assistance**
- **Priority Fixes**

For more information, [get a quote](mailto:enterprise@material-ui.com) from our team.

### Advisory

Supercharge your team’s development with Material-UI Advisory:

#### Developer Advisory

Everything your team needs to make sure you’re using React the best way as you evaluate and build on Material-UI.

> - Unlimited technical support covering Material-UI usage of the API, component design, performance, and architectural best practices.

Author: I have found this value proposing in Apollo offering, Ionic seems to offer the same, but use different wording. I would imagine that it can be sustainable at a high enough pricing point. Anyway, changing the chaning the wording:

> A private, dedicated Slack channel with access to our consultants.
  Get real-time advice, guidance, and feedback when you need it, with live Slack access and direct private chat with Material-UI’s expert technical team.

- A private, dedicated Slack channel with your team and Material-UI. You get access to our pool of Solutions Architect.
- An initial walkthrough of your architecture and development workflow.
- Guidance on best practices that help you take full advantage of Material-UI and prevent avoidable mistakes.

> - An early look at Material-UI features in the design and development phase.

Author: From Apollo, I wasn't too sure. Agree that it's not very useful. Removing it.

Content, tools, and support that can help you evangelize and teach React across your company.

[Get a quote](mailto:advisory@material-ui.com)

#### Enterprise Advisory

A complete support package, for teams using Material-UI for critical business applications and organizations deploying Material-UI across multiple teams.

- Includes "Developer Advisory" scaled to your entire scope of work.
- A designated Technical Account Manager who understands your business requirements and proactively makes sure you’re getting everything you need.
- A designated Solutions Architect who works with your team through every phase of development, on-line.
- Regular, scheduled reviews.

> - Prioritized assistance to help shepherd your open-source pull requests through the review process.

Reviewer: ```suggestion
```

Author: What issue do you see in this value proposition?

Author: The idea is that the "developer" offering focuses on making each dev get the most of the library while the "enterprise" offering focuses on making each organization the most of the library. In this case, what I had in mind is the following. Let's say an organization need something to be changed but doesn't want to wait for us to prioritize it, if it aligns with our direction, they could dedicate internal resources to the problem while we make sure we are not a bottleneck to make the changes land.

Ionic doesn't seem to have a similar value proposition. Maybe it's implicitly provided in the Enterprise offering. Under the "Priority Fixes" label.

Ok, I think that we can remove this dot.

[Get a quote](mailto:advisory@material-ui.com)

### Tidelift subscription <svg style="width: 24px; fill: currentColor; margin-left: 4px;" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 37 30.38"><g><g><path d="M16.07 0H2.77C0 0 0 0 0 2.54S0 5 2.77 5h7.91v9l5.39-4.2z"/><path d="M10.68 27.82c0 2.56 0 2.56 2.71 2.56s2.68 0 2.68-2.61V15l-5.39 4.22z"/><path d="M20.35 0h13.29C36.42 0 37 0 37 2.54S36.42 5 33.64 5h-7.9v9l-5.39-4.2z"/><path d="M25.74 27.82c0 2.56 0 2.56-2.71 2.56s-2.68 0-2.68-2.61V15l5.39 4.22z"/></g></g></svg>

Material-UI and the maintainers of thousands of other packages are working with Tidelift to deliver one enterprise subscription that covers all of the open-source you use.

If you want the flexibility of open-source and the confidence of commercial-grade software, this is worth looking at.

> The Tidelift Subscription manages your dependencies for you

Reviewer: ```suggestion
The Tidelift Subscription manages your dependencies for you:
```

- Get the tools you need to continuously catalog and understand the open-source software that your application depends on.

> - Your subscription helps pay the open-source community maintainers of the exact packages you use to ensure they meet the standards you require

Reviewer: ```suggestion
- Your subscription helps pay the open-source community maintainers of the packages you use, to ensure they meet the standards you require.
```

> - Address issues proactively, with tools that scan for new security, licensing, and maintenance issues, and alert our paticipating open-source maintainers so they can resolve them on your behalf.

Reviewer: ```suggestion
- Address issues proactively, with tools that scan for new security, licensing, and maintenance issues, and alert participating open-source maintainers so they can resolve them on your behalf.
```

> - Measure and improve your open-source dependencies health—which improves your app's health—and get a shortlist of high-impact steps your team can take to improve them even more.

Reviewer: ```suggestion
- Tidelift helps measure and improve your open-source dependencies' health – which improves your app's health – and gives a shortlist of high-impact steps your team can take to improve them even more.
```

> - Get commercial assurances that don't come for free with open-source packages, like intellectual property indemnification and support under a service level agreement. You expect these guarantees from proprietary software, and you can get them when using open-source as well.

Reviewer: ```suggestion
- Get commercial assurances that don't come for free with open-source packages, such as intellectual property indemnification and support under a service level agreement. You expect these guarantees from proprietary software, and you can get them when using open-source as well.
```

> The end result? All of the capabilities you expect from commercial-grade software, for the full breadth of open-source you use. That means less time grappling with esoteric open-source trivia, and more time building your own applications—and your business.

Reviewer: ```suggestion
The end result? All of the capabilities you expect from commercial-grade software, for the full breadth of open-source you use. That means less time grappling with esoteric open-source trivia, and more time building your own applications – and your business.
```

- **Version migration assistance**
- **Priority fixes**
- **Support**. Timely support and troubleshooting when you need it most. Get expert help directly from the Material-UI team with guaranteed response times (next-business day).

> - **Access to premium modules**. Datagrid, Charts, Scheduler, Rich Text Editor, Splitter, and more.

Reviewer: Doesn't exist, so shouldn't be offered...

Author: What about "Fake it until you make it“? cf https://marmelab.com/react-admin/

![Capture d’écran 2019-09-19 à 13 43 29](https://user-images.githubusercontent.com/3165635/65241299-7bbe2100-dae3-11e9-92f7-33046ac1e326.png)

Reviewer: You might get away with that in B2C (depending on the context), but not in B2B.

Also, most of those are months (or more) of of effort. So even if you "make it", you're now on the hook to deliver them before contract start date.

Reviewer 2: > What about "Fake it until you make it“?

This saying is not meant to be taken literally. It's used in the context of confidence about something or change in behavior. These components literally do not exist so it's not "fake it until you make it" but lying about the existence of something.

I'm pretty sure we can be immediately sued if someone buys that and there are no components. Even if we decide to make it a pre-order we're required to deliver it within a legally defined timeframe.

And even if you say "well they can't order it they have to contact first": How do you think most will react if they ask about those components and we don't have them? I don't think this will boost confidence in our product.

Author: Thanks for the feedback. I think that mentioning actual components is wrong. To be clear, the idea is to spark conversations with a bait so we can learn from people use cases. I don't think that we should try to sell something that doesn't exist yet, I also doubt a lot of companies will buy React components they can't use right away. 

I would propose a follow-up pull request where we only talk about "premium modules". I would also propose that we add a disclaimer in the roadmap.md that it's subject to change.

Would it be OK with you?

Reviewer: I'd suggest removing all mention, until such time as there is something to offer.

Author: How would we know what to build, and what first?
